CONTRIBUTING.md

```diff
@@ -28,3 +28,20 @@ make run
 ```sh
 make uninstall
 ```
+
+## Design choices
+
+### API
+The API of each OpenStack resource in Gopherkube is based on the corresponding OpenStack REST API. The name of the `snake_case` fields in the REST API are replaced with `camelCase`.
+
+### References
+Some resource definitions must refer to other resources. For example, a subnet definition refers to a network. In Gopherkube, a resource refers to another by the name of the corresponding Gopherkube manifest. For example: given an OpenStackNetwork with `metadata.name: network-1`, the OpenStackSubnet spec will have `spec.network: network-1` regardless of the UUID or the name of the network in the OpenStack cloud.
+
+### How to write a new CRD
+The `spec` of each OpenStack resource is based on the OpenStack REST API for creation (the POST call), with these modifications:
+* `snake_case` names must be replaced with their `camelCase` version;
+* fields that reference other OpenStack resources must reference the corresponding CRD by name. For example: in a subnet, the [REST API](https://docs.openstack.org/api-ref/network/v2/#create-subnet) requires the field `network_id`; the OpenStackSubnet CRD requires instead the field `network` that expects the name of the corresponding OpenStackNetwork.
+* optional fields are be represented as pointers. One exception is arrays, that are reresented as bare slices even when optional.
+* some resources can be further modified in the OpenStack API with additional calls REST API calls after creation. Gopherkube CRDs can embed the information needed to perform that second call, directly in the `spec`. For example, an OpenStackImage `spec` contains a `properties` array that the controller will use to set image properties with a separate PATCH call.
+
```

Comment:
IMHO it would be nice to empathize the need of validators, adding an extra bullet in this section or perhaps creating a whole new section for it

* Validations for each of the parameters that the OpenStack APIs expect occurs in the CRD definition. For this purpose, we rely on [Kubebuilder validations](https://book.kubebuilder.io/reference/markers/crd-validation.html).